base/sysimg.jl

```diff
@@ -96,7 +96,6 @@ include("subarray.jl")
 (::Type{Vector})() = Array{Any,1}(0)
```

Comment:
I would too but I have a suspicion it's fairly widely used, so I wanted to do the uncontroversial part first. It's fairly common to initialize a vector with zero elements and there was a period where `[]` created a Void eltype empty array. There's also a consistency argument here: if we deprecate that we should also deprecate `Vector{T}()` – are you in favor of that as well?